corehq/apps/users/models.py

```diff
@@ -1402,6 +1404,7 @@ def get_by_user_id(cls, userID, domain=None):
         if domain:
             if not couch_user.is_member_of(domain):
                 return None
+        cls.get_by_username.set_cached_value(couch_user.__class__, couch_user.username).to(couch_user)
```

Comment:
@dannyroberts curious to get your thoughts on this approach

Reply:
In general, I think it's great! Nice to see you using @pr33thi's new addition to quickcache :)

My reading is that there are two factors in determining whether this is a good idea:
1. Positive effect: Is a user that's looked up by id also likely to be looked up by username shortly thereafter (within 5 minutes) and vice versa for the similar line above?
2. Negative effect: Is the percentage of redis memory taken up by caching these functions significant?

If 1 then I think 2 matters less: we're happy to use up cache memory caching things that matter. If not 2, then 1 matters less: we're happy to do inexpensive caching even if we're not certain it makes a big difference. So I think this is a good idea if 1 or not 2 :)

Reply:
Thanks for your ever useful perspective. I'm pretty sure that 1 is true though I don't have any hard evidence of that. Certainly at the moment I think 2 is also true (at least on prod).

I think I'm going to keep it as is.